src/pysonar_scanner/configuration/pyproject_toml.py

```diff
@@ -37,23 +46,56 @@ def flatten_config_dict(config: dict[str, any], prefix: str) -> dict[str, any]:
     return result
 
 
-def load(base_dir: Path) -> Dict[str, str]:
+def load(base_dir: Path) -> TomlProperties:
     filepath = base_dir / "pyproject.toml"
     if not os.path.isfile(filepath):
-        return {}
+        return TomlProperties({}, {})
 
     try:
         with open(filepath, "rb") as f:
             toml_dict = tomli.load(f)
-
         # Look for configuration in the tool.sonar section
-        if "tool" in toml_dict and "sonar" in toml_dict["tool"]:
-            sonar_config = toml_dict["tool"]["sonar"]
-            python_to_java_names = {prop.python_name(): prop.name for prop in properties.PROPERTIES}
-            flattened_sonar_config = flatten_config_dict(sonar_config, prefix="sonar.")
-            return {python_to_java_names.get(key, key): value for key, value in flattened_sonar_config.items()}
-        return {}
+        sonar_properties = read_sonar_properties(toml_dict)
+        # Look for general project configuration
+        project_properties = read_project_properties(toml_dict)
+        return TomlProperties(sonar_properties, project_properties)
     except Exception:
-        # If there's any error parsing the TOML file, return empty dict
+        # If there's any error parsing the TOML file, return empty TomlProperties
         # SCANPY-135: We should log the pyproject.toml parsing error
-        return {}
+        return TomlProperties({}, {})
+
+
+def read_sonar_properties(toml_dict) -> Dict[str, str]:
+    if "tool" in toml_dict and "sonar" in toml_dict["tool"]:
+        sonar_config = toml_dict["tool"]["sonar"]
+        python_to_java_names = {prop.python_name(): prop.name for prop in properties.PROPERTIES}
+        flattened_sonar_config = flatten_config_dict(sonar_config, prefix="sonar.")
+        return {python_to_java_names.get(key, key): value for key, value in flattened_sonar_config.items()}
+    return {}
+
+
+def read_project_properties(toml_dict) -> Dict[str, str]:
+    # Extract project metadata
+    project_properties = {}
+    if "project" in toml_dict:
+        project_data = toml_dict["project"]
+        # Known pyproject.toml project keys and associated Sonar property names
+        property_mapping = {
+            "name": properties.SONAR_PROJECT_NAME,
+            "description": properties.SONAR_PROJECT_DESCRIPTION,
+            "version": properties.SONAR_PROJECT_VERSION,
+            "requires-python": properties.SONAR_PYTHON_VERSION,
+        }
+
+        for toml_key, sonar_property in property_mapping.items():
+            if toml_key in project_data:
+                string_property = convert_arrays_to_string(project_data[toml_key])
+                project_properties[sonar_property] = string_property
+    return project_properties
+
+
+def convert_arrays_to_string(property) -> str:
+    if isinstance(property, list):
+        return ",".join(str(item) for item in property)
+    else:
+        return property
```

Comment:
I would suggest regrouping those functions in a class, like *TomlLoader* or *TomlConfigurationLoader*. If you do so, You could rename also the file accordingly to the class name. I think it would be easier to understand like this.

Reply:
I regrouped the functions in a class, but I did not actually rename the file: I feel that the current naming is consistent (`cli`, `sonar_project_properties`, `pyproject_toml`, etc...).